### ISSUE TITLE
How to `#include` some headers in .gitignore

### ISSUE THREAD

**Issue by gtianw** [microsoft/vscode-cpptools#9528]
Bug type: Language Service

**Describe the bug**
- OS and Version:Windows 10 Pro
- VS Code Version:1.68.1
- C/C++ Extension Version:1.10.7
- Other extensions you installed (and if the issue persists after disabling them):None
- If using SSH remote, specify OS of remote machine:No

When I open the FFmpeg project, header files in &ldquo;.gitignore&rdquo; cannot be included. The problem may be associated with subdirectories.
Such as &ldquo;config.h&rdquo; and &ldquo;libavutil/avconfig.h&rdquo;, which are generated by `configure`. But these files, have important global symbols, must be included for parsing completely.

**Steps to reproduce**

**Expected behavior**

**Code sample and logs**

- Code sample

- Configurations in `c_cpp_properties.json`

- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette

- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)

**Screenshots**

- `#include "config.h"` in &ldquo;fftools/ffplay.c&rdquo;, without &ldquo;.gitignore&rdquo; in `fftools/`:
![image](https://user-images.githubusercontent.com/91190888/176867571-4c08e400-96c8-42c3-88bb-557bc053a0fd.png)
- `#include "config.h"` in &ldquo;libavutil/adler32.c&rdquo;, with a &ldquo;.gitignore&rdquo; in `libavutil/`:
![image](https://user-images.githubusercontent.com/91190888/176868208-44a0f2e8-6c4e-4c5e-9e22-9451b8ce212a.png)

**Additional context**

**Comment by sean-mcmanus** [comment]
Our extension does no processing of .gitignore, so I'm guessing the problem is that you just need to make sure the path to config.h is in your include path. In your 2 examples, you're using `#include ""` which will add the current folder to the includePath, which may be why the include is only found in files in certain folders. Can you check where config.h is located and that the includePath includes that location?

**Comment by gtianw** [comment]
@sean-mcmanus 
I make sure that config.h is in my includePath.
The following is source tree:
![image](https://user-images.githubusercontent.com/91190888/176985215-65e67261-bfe7-4df0-8dd6-43242208f6b6.png)
The following is my c_cpp_properties.json:
```json
{
    "configurations": [
        {
            "name": "Win64",
            "includePath": [
                "${workspaceFolder}/**",
                "C:\\Compiler\\msys2\\mingw64\\include\\SDL2\\"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE"
            ],
            "compilerPath": "C:/Compiler/msys2/mingw64/bin/gcc.exe",
            "cStandard": "c11",
            "cppStandard": "c++11",
            "intelliSenseMode": "${default}",
            "configurationProvider": "ms-vscode.makefile-tools"
        }
    ],
    "version": 4
}
```

**Comment by gtianw** [comment]
@sean-mcmanus 
When I delete "configurationProvider" in c_cpp_properties.json, all is well.
Maybe this is a compatibility problem between ms-vscode.makefile-tools and ms-vscode.cpptools.

**Comment by sean-mcmanus** [comment]
Yeah, if you have configurationProvider set, then we get the include paths from Makefile tools, so I'll tranfer this issue there -- it's possible they check .gitignore.
